dev-requirements.txt

```diff
@@ -4,4 +4,6 @@ pytest
 pytest-asyncio
 pytest-httpserver
 urllib3
-trustme
\ No newline at end of file
+trustme
+uvicorn
+xpresso
```

Comment:
We can make this just Starlette or event write a plain ASGI app (as long as we're okay with more boilerplate)

Reply:
In the interest of fewer test dependencies we should probably write ASGI or starlette. The app itself can live inside of a `conftest.py` if more space is needed.

Reply:
I can do you one better: you already have `aiohttp` installed, so we can use that as both the server and client. Zero extra dependencies.